core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -3197,4 +3200,192 @@ class KafkaApisTest {
+    //    handleAlterIsrRequest
+    //    handleEnvelope
+  }
+
+  @Test
+  def testRaftHandleUpdateMetadataRequest(): Unit = {
+    val request: RequestChannel.Request = EasyMock.createNiceMock(classOf[RequestChannel.Request])
+    val kafkaApis = createKafkaApis(raftSupport = true)
+    val e = assertThrows(classOf[UnsupportedVersionException],
+      () => kafkaApis.handleUpdateMetadataRequest(request))
+    assertEquals(kafkaApis.shouldNeverReceive(request).getMessage, e.getMessage)
+    //    handleControlledShutdownRequest
+    //    handleAlterIsrRequest
+    //    handleEnvelope
+  }
+
+  @Test
+  def testRaftHandleControlledShutdownRequest(): Unit = {
+    val request: RequestChannel.Request = EasyMock.createNiceMock(classOf[RequestChannel.Request])
+    val kafkaApis = createKafkaApis(raftSupport = true)
+    val e = assertThrows(classOf[UnsupportedVersionException],
+      () => kafkaApis.handleControlledShutdownRequest(request))
+    assertEquals(kafkaApis.shouldNeverReceive(request).getMessage, e.getMessage)
+    //    handleAlterIsrRequest
+    //    handleEnvelope
+  }
+
+  @Test
+  def testRaftHandleAlterIsrRequest(): Unit = {
+    val request: RequestChannel.Request = EasyMock.createNiceMock(classOf[RequestChannel.Request])
+    val kafkaApis = createKafkaApis(raftSupport = true)
+    val e = assertThrows(classOf[UnsupportedVersionException],
+      () => kafkaApis.handleAlterIsrRequest(request))
+    assertEquals(kafkaApis.shouldNeverReceive(request).getMessage, e.getMessage)
+  }
+
+  @Test
+  def testRaftHandleEnvelope(): Unit = {
+    val request: RequestChannel.Request = EasyMock.createNiceMock(classOf[RequestChannel.Request])
+    val kafkaApis = createKafkaApis(raftSupport = true)
+    val e = assertThrows(classOf[UnsupportedVersionException],
+      () => kafkaApis.handleEnvelope(request))
+    assertEquals(kafkaApis.shouldNeverReceive(request).getMessage, e.getMessage)
+  }
+
+  @Test
+  def testRaftHandleCreateTopicsRequest(): Unit = {
+    val request: RequestChannel.Request = EasyMock.createNiceMock(classOf[RequestChannel.Request])
+    val kafkaApis = createKafkaApis(raftSupport = true)
+    val e = assertThrows(classOf[UnsupportedVersionException],
+      () => kafkaApis.handleCreateTopicsRequest(request))
+    assertEquals(kafkaApis.shouldAlwaysBeForwarded(request).getMessage, e.getMessage)
+  }
+
+  @Test
+  def testRaftHandleCreatePartitionsRequest(): Unit = {
+    val request: RequestChannel.Request = EasyMock.createNiceMock(classOf[RequestChannel.Request])
+    val kafkaApis = createKafkaApis(raftSupport = true)
+    val e = assertThrows(classOf[UnsupportedVersionException],
+      () => kafkaApis.handleCreatePartitionsRequest(request))
+    assertEquals(kafkaApis.shouldAlwaysBeForwarded(request).getMessage, e.getMessage)
+  }
+
+  @Test
+  def testRaftHandleDeleteTopicsRequest(): Unit = {
+    val request: RequestChannel.Request = EasyMock.createNiceMock(classOf[RequestChannel.Request])
```

Comment:
nit: there's a lot of redundancy in these test cases. Could we define helpers? For example:
```scala
def assertUnsupportedWithRaft(handler: RequestChannel.Request => Unit)

assertUnsupportedWithRaft(kafkaApis.handleDeleteTopicsRequest)
```